apps/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -76,7 +123,40 @@ export class PackageChangeAnalyzer {
     return projectState;
   }
 
-  private async _getData(terminal: Terminal): Promise<Map<string, Map<string, string>> | undefined> {
+  /**
+   * Gets a list of projects that have changed in the current state of the repo
+   * when compared to the specified branch.
+   */
+  public async *getChangedProjectsAsync(
+    targetBranch: string,
+    shouldFetch: boolean = false
```

Comment:
Let's make `shouldFetch` into an options interface.  `getChangedProjectsAsync(branch, true)` isn't very intelligible nor is it extensible.